src/DesignCompile/CompileExpression.cpp

```diff
@@ -433,13 +433,13 @@ static bool largeInt(std::string value) {
     largeInt = true;
   } else if (value.size() == 20) {
     if (isSigned) {
-      int64_t test = std::strtoll(value.c_str(), 0, 10);
+      int64_t test = std::strtoll(value.data(), 0, 10);
```

Comment:
this is invalid, as value is not nul terminated (also in other places in this PR).

Use `std::from_chars()` instead if you want to parse values from string-views.

Reply:
Yes, converting first to std::from_chars() and then doing the string_view change sounds good.

Reply:
ah, that can be because std::from_char is optional and older compilers (we're using gcc 9 in the CI) can't deal with it.
I think we need to have a fallback situation in that case.

I've implemented one in another project, maybe we need such a thing in Surelog
https://github.com/hzeller/beagleg/blob/main/src/common/string-util.cc#L82-L157

Reply:
... and we do want to be able to use an slightly older compiler as many people in the EDA world unfortunately have pretty outdated machines.